src/qt/walletmodel.h

```diff
@@ -69,24 +71,49 @@ class SendCoinsRecipient
         std::string sAddress = address.toStdString();
         std::string sLabel = label.toStdString();
         std::string sMessage = message.toStdString();
+        std::string sRedeemScript = redeemScript.toStdString();
         std::string sPaymentRequest;
         if (!ser_action.ForRead() && paymentRequest.IsInitialized())
             paymentRequest.SerializeToString(&sPaymentRequest);
         std::string sAuthenticatedMerchant = authenticatedMerchant.toStdString();
 
+        if(!ser_action.ForRead() && (this->nVersion == 1) && !redeemScript.isEmpty()) {
+            this->nVersion = 2;
+        }
+
         READWRITE(this->nVersion);
         READWRITE(sAddress);
         READWRITE(sLabel);
         READWRITE(amount);
-        READWRITE(sMessage);
+
+        // Sort of a hack but allows backward compatibility:
```

Comment:
Since we're anyway updating the version, the only compatibility benefit I see is that older QT wallets can read the wallet file of newer versions, i.e. we're assuring forward compatibility. I'm not sure if that benefit outweighs clean code in this case? I'm thinking that it's better to just update the version statically and add the field to the tail?

Reply:
Yeah I'm not too keen on that piece either. We did break wallet backward compat once before when we updated bdb. I do remember some cases where people needed to go through bdb tools to revert that cause they needed to revert to an older client version. I think that was mostly due to performance/resource usage issues with the new version. Not 100% sure on the exact point that happened. Might have been 1.7 to 1.10.

Either way this leads me to believe that backwards compat is still somewhat important and should be considered. What would happen if you'd try to open a "new" wallet on an older version? Is there a path back?